src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -72,6 +72,24 @@ internal sealed class LocalFunctionSymbol : MethodSymbol
                 ReportAttributesDisallowed(param.AttributeLists, _declarationDiagnostics);
             }
 
+            if (syntax.ReturnType.Kind() == SyntaxKind.RefType)
+            {
+                var returnType = (RefTypeSyntax)syntax.ReturnType;
+                if (returnType.ReadOnlyKeyword.Kind() == SyntaxKind.ReadOnlyKeyword)
+                {
+                    _refKind = RefKind.RefReadOnly;
+                    DeclaringCompilation.EnsureIsReadOnlyAttributeExists(_declarationDiagnostics, returnType.Location, modifyCompilation: false);
+                }
+                else
+                {
+                    _refKind = RefKind.Ref;
+                }
+            }
+            else
+            {
+                _refKind = RefKind.None;
+            }
+
             _binder = binder;
```

Comment:
>if (syntax.ReturnType.Kind() == SyntaxKind.RefType) [](start = 12, length = 51)

I think this ```if``` can stay, we simply shouldn't call DeclaringCompilation.EnsureIsReadOnlyAttributeExists inside it.